crates/autopilot/src/solvable_orders.rs

```diff
@@ -348,6 +313,66 @@ impl SolvableOrdersCache {
         Ok(())
     }
 
+    async fn fetch_balances(&self, queries: Vec<Query>) -> HashMap<Query, U256> {
+        let fetched_balances = {
+            let _timer = self.stage_timer("balance_fetch");
+            self.balance_fetcher.get_balances(&queries).await
+        };
+        queries
+            .into_iter()
+            .zip(fetched_balances)
+            .filter_map(|(query, balance)| match balance {
+                Ok(balance) => Some((query, balance)),
+                Err(err) => {
+                    tracing::warn!(
+                        owner = ?query.owner,
+                        token = ?query.token,
+                        source = ?query.source,
+                        error = ?err,
+                        "failed to get balance"
+                    );
+                    None
+                }
+            })
+            .collect()
+    }
+
+    /// Executed effectful orders filtering in parallel.
```

Comment:
What does "effectful" in mean here?

Reply:
I tried to express "side effect" operations in other words. Removed to avoid confusion.